src/extensions/FileUploadExtension.ts

```diff
@@ -75,11 +76,27 @@ export const FileUploadExtension = Extension.create<FileUploadOptions>({
         props.tr.setMeta('uploadFiles', true)
         return true
       },
+      getMetaTags: () => (props: CommandProps) => {
+        const tags: string[][] = []
+        // make sure the file uploaded is still in the editor content
+        props.editor.state.doc.descendants((node) => {
```

Comment:
I am not so sure on what you trying to do here, you are iterating on all documents looking for image and video nodes, when you already have them on `props.editor.storage.fileUpload.files`

Reply:
The reason I am iterating here is to check if the file is still in the editor.

When you are using immediateUpload, maybe that file has been subsequently deleted, the deleted action is not tracked in the extension is it?

Reply:
You are correct, I don't think we should manually populate the storage files as we can't track when the user deletes an image, we should follow the same approach of NostrExtension storage get functions with something like `getFiles()` https://github.com/cesardeazevedo/nostr-editor/blob/main/src/extensions/NostrExtension.ts#L139-L182